packages/opentelemetry-core/src/context/propagation/HttpTraceContext.ts

```diff
@@ -41,19 +47,33 @@ const VERSION = '00';
  *     For more information see {@link https://www.w3.org/TR/trace-context/}
  */
 export function parseTraceParent(traceParent: string): SpanContext | null {
-  const match = traceParent.match(VALID_TRACE_PARENT_REGEX);
+  const trimmed = traceParent.trim();
+  const traceParentParts = trimmed.split('-');
+
+  // Current version must be structured correctly.
+  // For future versions, we can grab just the parts we do support.
   if (
-    !match ||
-    match[1] === '00000000000000000000000000000000' ||
```

Comment:
These validations moved into the respective regex validators.

Reply:
+1